src/coreclr/nativeaot/Runtime/wasm/PInvoke.cpp

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <stdlib.h>
+
+#include "CommonTypes.h"
+#include "CommonMacros.h"
+
+extern "C" void* t_pShadowStackBottom = nullptr;
+extern "C" void* t_pShadowStackTop;
+
+COOP_PINVOKE_HELPER(void*, RhpGetOrInitShadowStackTop, ())
```

Comment:
This inits the `t_pShadowStackBottom` but the name reflects the logical function rather than the implementation?

Reply:
Yes... I renamed this a few times, happy to try more options if they are clearer, e. g. `RhpGetShadowStackTopWithInit`.

The semantics of these helpers are tied very strongly to what the code generator does, so the fact we don't set `t_pShadowStackTop` is because we know it would be a dead store.

Reply:
I think its fine, nothing better comes to mind really.